Doc/using/cmdline.rst

```diff
@@ -994,3 +994,117 @@ Debug-mode variables
+
+The :envvar:`PYTHONHOME` environment variable may be used to set the ``prefix``
+and ``exec_prefix`` locations. Otherwise these directories are found by using
+the Python executable as a starting point and then looking for various 'landmark'
+files and directories. Note that any symbolic links are followed so the real
+Python executable location is used as the search starting point. The Python
+executable location is called ``home``.
+
+Once ``home`` is determined, the ``prefix`` directory is found by first looking
+for :file:`python{majorversion}{minorversion}.zip` (``python311.zip``). On Windows
+the zip archive is searched for in ``home`` and on Unix the archive is expected
+to be in :file:`lib`. Note that the expected zip archive location is added to the
+module search path even if the archive does not exist. If no archive was found,
+Python on Windows will continue the search for ``prefix`` by looking for :file:`Lib\\os.py`.
+Python on Unix will look for :file:`lib/python{majorversion}.{minorversion}/os.py`
+(``lib/python3.11/os.py``). On Windows ``prefix`` and ``exec_prefix`` are the same,
+however on other platforms :file:`lib/python{majorversion}.{minorversion}/lib-dynload`
+(``lib/python3.11/lib-dynload``) is searched for and used as an anchor for
+``exec_prefix``.
+
+Once found, ``prefix`` and ``exec_prefix`` may be accessed at :data:`sys.prefix`
+and :data:`sys.exec_prefix`.
+
+Finally, the :mod:`site` module is processed and :file:`site-packages` directories
+are added to the module search path. A common way to customize the search path is
+to create :mod:`sitecustomize` or :mod:`usercustomize` modules as described in
+the :mod:`site` module documentation.
+
+.. note::
+
+   The effect of :mod:`site` on the module search path may be seen by running
+   Python with ``-S`` which will start Python without importing site. ``-s`` will
+   skip adding the :data:`user site-packages directory <site.USER_SITE>` to :data:`sys.path`.
+
+   The effect of :envvar:`PYTHON*` environment variables may be seen by running
+   Python with ``-E`` which will start Python ignoring these environment variables.
+
+   Finally, ``-I`` will start Python in isolated mode, simlar to ``-E`` and ``-s``
+   but with further restrictions.
+
+Virtual environments
+~~~~~~~~~~~~~~~~~~~~
+
+If Python is run in a virtual environment (as described at :ref:`tut-venv`)
+then ``prefix`` and ``exec_prefix`` are specific to the virtual environment.
+
+If a ``pyvenv.cfg`` file is found alongside the main executable, or in the
+directory one level above the executable, the following variations apply:
+
+* If ``home`` is an absolute path and :envvar:`PYTHONHOME` is not set, this
+  path is used instead of the path to the main executable when deducing ``prefix``
+  and ``exec_prefix``.
+
+_pth files
+~~~~~~~~~~
+
+To completely override :data:`sys.path`, create a ``._pth`` file with the same
+name as the extension module (``python311._pth``) or the executable (``python._pth``)
```

Comment:
I think there is some confusion here: I believe the `python311` here is referring to the Python interpreter shared library (a DLL), not an extension module (confusingly also a DLL). @zooba, can you confirm and perhaps suggest a more correct wording?

Reply:
Also, it this a Windows only feature? If not, the wording needs to take into account the fact that on Unix-y systems the python executable might have a shared library component.

Reply:
It should be the same name as and located beside the Python executable or shared library (not "extension module") -- e.g. "python.exe" -> "python._pth" or "python311.dll" -> "python311._pth". 

Support for "._pth" files was extended to all platforms. I've tested this in Linux for a "._pth" file that's based on the real executable name, as well as a symlink to the executable, as well as a symlink to the "._pth" file. All work, but note that relative paths are resolved against the opened location of the "._pth" file. The real path of the "._pth" file is not used. I haven't checked how a "._pth" file works when named for a shared library, if the interpreter is built as such. Apparently using the shared-library path [isn't tested](https://github.com/python/cpython/blob/v3.11.0a6/Lib/test/test_site.py#L562) for POSIX.

Reply:
Thank you, I have been mixing my shared library and extension module terms far too freely. Will review.